### PR TITLE
add unit testing

### PR DESCRIPTION
**Pull request by aaditgupta21**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -11,3 +10,9 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-go@v2
       - run: sh misc/actions.sh 
+  testing:
```
Comment (safinsingh) [inline .github/workflows/build.yml:13]: `setup-go` already gets deps

**Review by safinsingh — CHANGES_REQUESTED** [review]
Actions already has a template for go projects, we should use that. i'll open a pr for that.

File: .github/workflows/build.yml (hunk 1 above)
Comment (aaditgupta21) [reply .github/workflows/build.yml:13]: This is a different job

Comment (safinsingh) [reply .github/workflows/build.yml:13]: see #37

**Pull request closed without merging**